### PR TITLE
Add the all comments list

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #314 :*
https://github.com/bcgov/met-public/issues/#314

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/common/Table/index.tsx [hunk 1]
```diff
@@ -100,8 +100,15 @@ interface MetTableProps<T> {
     headCells: HeadCell<T>[];
     defaultSort: keyof T;
     rows: T[];
+    hideHeader: boolean;
 }
-function MetTable<T>({ filter = { key: '', value: '' }, headCells = [], defaultSort, rows = [] }: MetTableProps<T>) {
+function MetTable<T>({
+    hideHeader,
```
Comment (jadmsaadaot) [inline met-web/src/components/common/Table/index.tsx:106]: make it default to false. hideHeader = false, this way you don't have to do <MetTable hideHeader={false}. You would only need to make ={true} when you need it

File: met-web/src/components/common/Table/index.tsx [hunk 2]
```diff
@@ -147,13 +154,18 @@ function MetTable<T>({ filter = { key: '', value: '' }, headCells = [], defaultS
...
                 <TableContainer>
                     <Table aria-labelledby="Engagements">
-                        <MetTableHead
-                            order={order}
-                            orderBy={orderBy}
-                            onRequestSort={handleRequestSort}
-                            rowCount={filteredRows.length}
-                            headCells={headCells}
-                        />
+                        {hideHeader ? (
```
Comment (jadmsaadaot) [inline met-web/src/components/common/Table/index.tsx:157]: try using <ConditionalComponent condition={!hideHeader}/> here

File: met-web/src/components/survey/listing/index.tsx [hunk 3]
```diff
@@ -169,7 +169,7 @@ const SurveyListing = () => {
                 </Button>
             </Grid>
             <Grid item xs={12} lg={10}>
-                <MetTable headCells={headCells} rows={surveys} defaultSort={'created_date'} />
+                <MetTable hideHeader={false} headCells={headCells} rows={surveys} defaultSort={'created_date'} />
```
Comment (jadmsaadaot) [inline met-web/src/components/survey/listing/index.tsx:172]: as mentioned above. Make hideaHeader default to false and remove this

File: met-web/src/routes/AuthenticatedRoutes.tsx [hunk 4]
```diff
@@ -21,6 +22,7 @@ const AuthenticatedRoutes = () => {
             <Route path="/engagement/form/:engagementId" element={<EngagementForm />} />
             <Route path="/engagement/view/:engagementId" element={<EngagementView />} />
             <Route path="/survey/:surveyId/comments" element={<CommentListing />} />
+            <Route path="/comments" element={<AllComments />} />
```
Comment (jadmsaadaot) [inline met-web/src/routes/AuthenticatedRoutes.tsx:25]: Also should be part of survey route and requires surveyId

File: met-web/src/components/comments/AllComments.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+                    </Grid>
+                    <Grid item xs={12}>
+                        <Typography sx={{ fontSize: '16px' }}>
+                            <b>Reviewed By: </b> {row.reviewed_by}
+                        </Typography>
+                    </Grid>
+                    <Grid item xs={6}></Grid>
+                    <Grid item xs={6}>
+                        {row.status != 'Approved' ? (
+                            <Box
```
[the remark this answers is not in the document]
Comment (jadmsaadaot) [reply met-web/src/components/comments/AllComments.tsx:111]: Always check MUI first

**Review by jadmsaadaot — APPROVED** [review]

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**